packages/health/android/src/main/kotlin/cachet/plugins/health/HealthPlugin.kt

```diff
@@ -1412,6 +1427,16 @@ class HealthPlugin(private var channel: MethodChannel? = null) :
         }
         binding.addActivityResultListener(this)
         activity = binding.activity
+
+
+        if ( healthConnectAvailable) {
+            val requestPermissionActivityContract = PermissionController.createRequestPermissionResultContract()
+
+            healthConnectRequestPermissionsLauncher =(activity as ComponentActivity).registerForActivityResult(requestPermissionActivityContract) { granted ->
```

Comment:
Shouldn't we use `FlutterFragmentActivity` instead of `ComponentActivity`

Reply:
We need the `ComponentActivity` in order to use the new api function `registerForActivityResult`, `FlutterFragmentActivity` is built upon the new Android apis which can be casted to a `ComponentActivity`. If you try to cast `FlutterActivity` to a `ComponentActivity` it will throw an error.